### ISSUE TITLE
Can't get token of a YLKG07YL with the script

### ISSUE THREAD

**Issue by Stooovie** [custom-components/ble_monitor#1254]
I'm unable to get token of my YLKG07YL with the get_beacon_key.py script. I run it and get this:

```
Activate pairing on your 'F8:24:41:C9:0E:57' device, then press Enter: 
Connection in progress...
Traceback (most recent call last):
  File "/home/pi/get_beacon_key.py", line 159, in <module>
    main(sys.argv)
  File "/home/pi/get_beacon_key.py", line 155, in main
    get_beacon_key(mac, product_id)
  File "/home/pi/get_beacon_key.py", line 106, in get_beacon_key
    peripheral = Peripheral(deviceAddr=mac)
  File "/home/pi/.local/lib/python3.9/site-packages/bluepy/btle.py", line 391, in __init__
    self._connect(deviceAddr, addrType, iface)
  File "/home/pi/.local/lib/python3.9/site-packages/bluepy/btle.py", line 439, in _connect
    raise BTLEDisconnectError("Failed to connect to peripheral %s, addr type: %s" % (addr, addrType), rsp)
bluepy.btle.BTLEDisconnectError: Failed to connect to peripheral F8:24:41:C9:0E:57, addr type: public
```
I DO push the button to enter the pairing mode, but it eventually crashes anyway.

the MAC is correct (got from hcitool lescan or Token extractor after pairing with the modded Mi home APK). 

I did get a token by pairing in the modded APK+the Token extractor app, but even with it I can't get anything else than the RSSI strenght to come through. At my wits' end. Any more tips?

**Comment by frickeo** [comment]
@Stooovie,
you might need to issue a `bluetoothctl power off` and `bluetoothctl power on` before using the script, especially if running on a raspberry pi. Your stacktrace shows that the initial connection fails (no _Successful connection!_ output).

Nevertheless, I later got a slightly different error message during the authentication process:
```
pi@piberry:~/ble $ python3 ./get_beacon_key.py DC:97:58:39:D6:D7 950
Activate pairing on your 'DC:97:58:39:D6:D7' device, then press Enter:
Connection in progress...
Successful connection!
Authentication in progress...
Traceback (most recent call last):
  File "/home/pi/ble/./get_beacon_key.py", line 159, in <module>
    main(sys.argv)
  File "/home/pi/ble/./get_beacon_key.py", line 155, in main
    get_beacon_key(mac, product_id)
  File "/home/pi/ble/./get_beacon_key.py", line 113, in get_beacon_key
    peripheral.writeCharacteristic(HANDLE_AUTH_INIT, MI_KEY1, "true")
  File "/home/pi/.local/lib/python3.9/site-packages/bluepy/btle.py", line 543, in writeCharacteristic
    return self._getResp('wr')
  File "/home/pi/.local/lib/python3.9/site-packages/bluepy/btle.py", line 407, in _getResp
    resp = self._waitResp(wantType + ['ntfy', 'ind'], timeout)
  File "/home/pi/.local/lib/python3.9/site-packages/bluepy/btle.py", line 368, in _waitResp
    raise BTLEGattError("Bluetooth command failed", resp)
bluepy.btle.BTLEGattError: Bluetooth command failed (code: 3, error: Attribute can't be written)
```

I solved this by modifying the script to use `bleak` instead of `bluepy`, see https://github.com/custom-components/ble_monitor/pull/1259

The modified get_beacon_key.py script is running on my raspi as well as on windows machine.

**Comment by Ernst79** [comment]
I've merged the PR #1259, so the old script is now replaced with the 'bleak' script of @frickeo. Thanks for that.
